libvast/src/view.cpp

```diff
@@ -293,4 +294,28 @@ bool evaluate_view(const data_view& lhs, relational_operator op,
   }
 }
 
+data_view to_canonical(const type& t, const data_view& x) {
+  auto v = detail::overload(
+    [](const view<enumeration>& x, const enumeration_type& t) -> data_view {
+      if (materialize(x) >= t.fields.size())
+        return caf::none;
+      return make_view(t.fields[materialize(x)]);
+    },
+    [&](auto&, auto&) { return x; });
+  return caf::visit(v, x, t);
+}
+
+data_view to_internal(const type& t, const data_view& x) {
+  auto v = detail::overload(
+    [](const view<std::string>& s, const enumeration_type& t) -> data_view {
```

Comment:
You could replace `const view<T>&` with `view<T>` for these two functions, but I'm not against leaving it as is.

Reply:
I left it as it was for now because `type_check()` and `evaluate_view()` do it this way, and I prefer consistency.